applications/nrf5340_audio/src/bluetooth/bt_management/bt_mgmt.h

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#ifndef _BT_MGMT_H_
+#define _BT_MGMT_H_
+
+#include <zephyr/bluetooth/bluetooth.h>
+#include <zephyr/bluetooth/conn.h>
+
+#define LE_AUDIO_EXTENDED_ADV_NAME                                                                 \
+	BT_LE_ADV_PARAM(BT_LE_ADV_OPT_EXT_ADV | BT_LE_ADV_OPT_USE_NAME,                            \
+			CONFIG_BLE_ACL_EXT_ADV_INT_MIN, CONFIG_BLE_ACL_EXT_ADV_INT_MAX, NULL)
+
+#define LE_AUDIO_EXTENDED_ADV_CONN_NAME                                                            \
+	BT_LE_ADV_PARAM(BT_LE_ADV_OPT_EXT_ADV | BT_LE_ADV_OPT_CONNECTABLE |                        \
+				BT_LE_ADV_OPT_USE_NAME,                                            \
+			CONFIG_BLE_ACL_EXT_ADV_INT_MIN, CONFIG_BLE_ACL_EXT_ADV_INT_MAX, NULL)
+
+#define LE_AUDIO_PERIODIC_ADV                                                                      \
+	BT_LE_PER_ADV_PARAM(CONFIG_BLE_ACL_PER_ADV_INT_MIN, CONFIG_BLE_ACL_PER_ADV_INT_MAX,        \
+			    BT_LE_PER_ADV_OPT_NONE)
+
+#define BT_LE_ADV_FAST_CONN                                                                        \
+	BT_LE_ADV_PARAM(BT_LE_ADV_OPT_CONNECTABLE, BT_GAP_ADV_FAST_INT_MIN_1,                      \
+			BT_GAP_ADV_FAST_INT_MAX_1, NULL)
+
+/* Broadcast name can max be 32 bytes long, so this will be the limit for both,
+ * add one for '\0' at the end
+ */
+#define BLE_SEARCH_NAME_MAX_LEN 33
+
+#if (CONFIG_SCAN_MODE_ACTIVE)
+#define NRF5340_AUDIO_GATEWAY_SCAN_TYPE	  BT_LE_SCAN_TYPE_ACTIVE
+#define NRF5340_AUDIO_GATEWAY_SCAN_PARAMS BT_LE_SCAN_ACTIVE
+#elif (CONFIG_SCAN_MODE_PASSIVE)
+#define NRF5340_AUDIO_GATEWAY_SCAN_TYPE	  BT_LE_SCAN_TYPE_PASSIVE
+#define NRF5340_AUDIO_GATEWAY_SCAN_PARAMS BT_LE_SCAN_PASSIVE
+#else
+#error "Please select either CONFIG_SCAN_MODE_ACTIVE or CONFIG_SCAN_MODE_PASSIVE"
+#endif
+
+enum bt_mgmt_scan_type {
+	BT_MGMT_SCAN_TYPE_CONN = 1,
+	BT_MGMT_SCAN_TYPE_BROADCAST = 2,
+};
+
+/**
+ * @brief	Start scanning for advertisements.
+ *
+ * @param	scan_intvl	Scan interval in units of 0.625ms.
+ *				Can be 0, valid range 0x4 - 0xFFFF.
+ * @param	scan_win	Scan window in units of 0.625ms.
+ *				Can be 0, valid range 0x4 - 0xFFFF.
+ * @param	type		Type to scan for (conn, broadcast, etc).
+ * @param	name		Name to search for, device name or broadcast name,
+ *				depending on type of search. Can max be
+ *				BLE_SEARCH_NAME_MAX_LEN long, everything beyond that
+ *				will be cropped. Can be NULL.
+ *
+ * @note	If 0 or NULL is given as the input the previous/default value will be used.
```

Comment:
It depends on the type of parameter and is stated per parameter above, if it says `can be 0` or `can be NULL`

Reply:
Yes, that I get, but how the sentence sounds now, the value to be used will be previous/default regardless of whether 0 or NULL is given -- it will be previous/default in both cases. So my question is: does 0 correspond to previous value and NULL to default value?

Side note: previous/default -- by using / here, you're basically saying "previous or default, or both", so the meaning of the current sentence is even more confusing, as it suggests that the value will be selected at random.

Reply:
NULL is only valid for pointers (which only name is) and 0 is only valid for int (which is scan_intvl and scan_win).

Reply:
Then it's even more confusing for me to see on what the choice of previous or default depends.

Reply:
Clarified (hopefully):
```
 * @note	To restart scanning, call this function with all 0s and NULL, except for @p type.
 *		The same scanning parameters as when bt_mgmt_scan_start was first called will then
 *		be used.
```